src/pages/myAppointments/[appointment_id].tsx

```diff
@@ -0,0 +1,66 @@
+import React, { useContext, useEffect, useState } from "react";
+import Navbar from "@/components/public/Navbar";
+import { Box, Link } from "@mui/material";
+import ArrowBackIcon from "@mui/icons-material/ArrowBack";
+import AppointmentParticipantCard from "@/components/appointment/AppointmentParticipantCard";
+import AppointmentHostCard from "@/components/appointment/AppointmentHostCard";
+import { AppointmentDetail } from "@/types/Appointment";
+import { GetAppointmentsByAppointmentId } from "@/services/Appointment";
+import { useRouter } from "next/router";
+import { useSupabaseClient } from "@supabase/auth-helpers-react";
+import { Database } from "supabase/db_types";
+import { PagePaths } from "enum/pages";
+import Loading from "@/components/public/Loading";
+import { userContext } from "supabase/user_context";
+
+export default function Home() {
+  const router = useRouter();
+  const userStatus = useContext(userContext);
+  const supabaseClient = useSupabaseClient<Database>();
+  const [appointment, setAppointment] = useState<AppointmentDetail | null>(null);
+  const [isHost, setIsHost] = useState<boolean | null>(null);
+  const [isParticipant, setIsParticipant] = useState<boolean | null>(null);
+
+  const appointmentId = parseInt(router.query.appointment_id as string);
+  useEffect(() => {
+    if (!appointmentId || !userStatus.user) return;
+    GetAppointmentsByAppointmentId(appointmentId, supabaseClient).then((appointment) => {
+      setIsHost(userStatus.user?.userId == appointment?.ownerId);
+      setIsParticipant(appointment?.acceptParticipants.includes(userStatus.user?.username as string))
+      setAppointment(appointment);
+    }).catch((err) => {
+      console.log(err)
+      return;
+    })
+
+  }, [supabaseClient, appointmentId, userStatus.user?.userId]);
+
+  function handleGoBack(): void {
+    router.push(PagePaths.myAppointments);
+    return;
+  }
+  if (!appointment) return <Loading />
+  if (!isParticipant) return handleGoBack()
```

Comment:
Need to add guard for user that hasn't logged in.

Reply:
done fixed it